addons/medical_treatment/functions/fnc_onMedicationUsage.sqf

```diff
@@ -1,68 +1,50 @@
 #include "..\script_component.hpp"
 /*
- * Author: Glowbal
+ * Author: Glowbal, Cplhardcore
  * Handles the medication given to a patient.
  *
  * Arguments:
- * 0: The patient <OBJECT>
+ * 0: Patient <OBJECT>
  * 1: Medication Treatment classname <STRING>
- * 2: Max dose (0 to ignore) <NUMBER>
- * 3: Max dose deviation <NUMBER>
- * 3: Incompatable medication <ARRAY<STRING>>
+ * 2: Incompatible medication <ARRAY of <STRING, NUMBER>>
  *
  * Return Value:
  * None
  *
  * Example:
- * [player, "morphine", 4, 2, [["x", 1]]] call ace_medical_treatment_fnc_onMedicationUsage
+ * [player, "morphine", [["x", 1]]] call ace_medical_treatment_fnc_onMedicationUsage
  *
  * Public: No
  */
 
-params ["_target", "_className", "_maxDose", "_maxDoseDeviation", "_incompatibleMedication"];
-TRACE_5("onMedicationUsage",_target,_className,_maxDose,_maxDoseDeviation,_incompatibleMedication);
-
-private _overdosedMedications = [];
+params ["_target", "_className", "_incompatibleMedication"];
+TRACE_3("onMedicationUsage",_target,_className,_incompatibleMedication);
 
 // Check for overdose from current medication
+private _defaultConfig    = configFile >> QUOTE(ADDON) >> "Medication";
+private _medicationConfig = _defaultConfig >> _classname;
+private _maxDose          = GET_NUMBER(_medicationConfig >> "maxDose",getNumber (_defaultConfig >> "maxDose"));
+
 if (_maxDose > 0) then {
-    private _currentDose = [_target, _className] call EFUNC(medical_status,getMedicationCount);
+    private _maxDoseDeviation = GET_NUMBER(_medicationConfig >> "maxDoseDeviation",getNumber (_defaultConfig >> "maxDoseDeviation"));
+    private _currentDose = [_target, _className] call EFUNC(medical_status,getMedicationCount) select 1;
     // Because both {floor random 0} and {floor random 1} return 0
     if (_maxDoseDeviation > 0) then {
         _maxDoseDeviation = _maxDoseDeviation + 1;
     };
 
-    if (_currentDose > _maxDose + (floor random _maxDoseDeviation)) then {
+    private _limit = _maxDose + (floor random _maxDoseDeviation);
+    if (_currentDose > _limit) then {
         TRACE_1("exceeded max dose",_currentDose);
-        _overdosedMedications pushBackUnique _className;
+        [_target, _classname, _currentDose, _limit, _classname] call FUNC(overDose);
     };
 };
 
 // Check incompatible medication (format [med,limit])
 {
     _x params ["_xMed", "_xLimit"];
     private _inSystem = [_target, _xMed] call EFUNC(medical_status,getMedicationCount);
```

Comment:
This needs to be updated, as `EFUNC(medical_status,getMedicationCount)` returns an array now.

Reply:
In fact, there are two other locations that need to be updated:
- `medical_statemachine\functions\fnc_handleStateUnconscious.sqf`
- `medical\dev\watchVariable.sqf`